pkg/scheduler/framework/preemption/preemption.go

```diff
@@ -340,9 +343,26 @@ func (ev *Evaluator) prepareCandidate(ctx context.Context, c Candidate, pod *v1.
 		// Otherwise we should delete the victim.
 		if waitingPod := fh.GetWaitingPod(victim.UID); waitingPod != nil {
 			waitingPod.Reject(pluginName, "preempted")
-		} else if err := util.DeletePod(ctx, cs, victim); err != nil {
-			klog.ErrorS(err, "Preempting pod", "pod", klog.KObj(victim), "preemptor", klog.KObj(pod))
-			return framework.AsStatus(err)
+		} else {
+			if feature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
+				condition := &v1.PodCondition{
+					Type:    v1.AlphaNoCompatGuaranteeDisruptionTarget,
+					Status:  v1.ConditionTrue,
+					Reason:  "PreemptionByKubeScheduler",
+					Message: "Kube-scheduler: preempting",
+				}
+				newStatus := pod.Status.DeepCopy()
+				if apipod.UpdatePodCondition(newStatus, condition) {
+					if err := util.PatchPodStatus(ctx, cs, victim, newStatus); err != nil {
+						klog.ErrorS(err, "Preparing pod preemption", "pod", klog.KObj(victim), "preemptor", klog.KObj(pod))
+						return framework.AsStatus(err)
```

Comment:
naive question since I'm not familiar with this bits, I did a similar one before,  before the pod was always deleted, but now it seems that if the patch to the pod status fail the pod is never deleted ... is that expected?

Reply:
That is correct. The scheduler might want to delete the same pod later. But it might decide to delete a different pod or no pod at all. That's why we will clear the condition if this happens #111475

Reply:
Without the patch the delete could also fail - say due to api server being temporarily down. However, indeed, with patches we risk conflicts, which could fail more often than simple deletes. @alculquicondor  @soltysh opinions?

Reply:
We could proceed with deletes if the patch fails, but I suggest to re-evaluate the decision for Beta. Anyway for Beta we are planning to commonize the the code for appending the pod conditions.